packages/app/docs/stories/appWithReduxMiddleware.story.js

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+import { Button, Card } from 'react-rainbow-components';
+import { useDispatch, useSelector } from 'react-redux';
+import { w3cwebsocket as W3CWebSocket } from 'websocket';
+import app from '../../../../firebase';
+import RainbowFirebaseApp from '../../src/components/App';
+
+const getIndicatorColor = (props) => {
+    if (props.isConnected) return props.palette.success.main;
+    return props.palette.error.main;
+};
+
+const Container = styled.div`
+    padding: 72px;
+`;
+
+const ConnectionStatus = styled.div.attrs((props) => props.theme.rainbow)`
+    width: 1.5rem;
+    height: 1.5rem;
+    border-radius: 0.875rem;
+    background: ${(props) => getIndicatorColor(props)};
+`;
+
+class SocketConnection {
+    constructor() {
+        this.socket = null;
+    }
+
+    connect(endpoint, handlers) {
+        const { onConnect, onDisconnect, onError, onMessage } = handlers;
+        if (this.isConnected()) {
+            this.socket.close();
+        }
+        this.socket = W3CWebSocket(endpoint);
+        this.socket.onopen = onConnect;
+        this.socket.onerror = onError;
+        this.socket.onclose = onDisconnect;
+        this.socket.onmessage = onMessage;
+    }
+
+    disconnect() {
+        if (this.isConnected()) {
+            this.socket.close();
+        }
+        this.socket = null;
+    }
+
+    send(data) {
+        if (this.isConnected()) {
+            this.socket.send(data);
+        }
+    }
+
+    isConnected() {
+        return this.socket !== null;
+    }
+}
+
+export const appWithReduxMiddleware = () => {
+    const init = { isConnected: false, isBussy: false };
+    const websocketReducer = (state = init, action) => {
+        switch (action.type) {
+            case 'CONNECTING':
+                return {
+                    ...init,
+                    isBussy: true,
+                };
+            case 'CONNECTED':
+                return {
+                    ...init,
+                    isConnected: true,
+                };
+            case 'DISCONNECTING':
+                return {
+                    ...state,
+                    isBussy: true,
+                };
+            case 'DISCONNECTED':
+                return {
+                    ...init,
+                };
+            default:
+                return state;
+        }
+    };
+
+    const websocketMiddleware = () => {
+        const socket = new SocketConnection();
+
+        const onSocketConnect = (store) => () => {
+            store.dispatch({ type: 'CONNECTED' });
+        };
+
+        const onSocketClose = (store) => () => {
+            store.dispatch({ type: 'DISCONNECTED' });
+        };
+
+        const onError = (store) => () => {
+            store.dispatch({ type: 'DISCONNECTED' });
+        };
```

Comment:
while staying connected if I switch off the internet of my PC the UI remains as connected, I think it should react since the socket connection should know there is not internet